scripts/watchdog

```diff
@@ -0,0 +1,149 @@
+    ONIE_PLATFORM="$(cat /host/machine.conf | grep 'onie_platform=' | cut -f2 -d'=')"
+    ABOOT_PLATFORM="$(cat /host/machine.conf | grep 'aboot_platform=' | cut -f2 -d'=')"
+    ONIE_SWITCH_ASIC="$(cat /host/machine.conf | grep 'onie_switch_asic=' | cut -f2 -d'=')"
+}
+
+function isDellS6100() {
+    local -r _DELLEMC_S6100_PLATFORM="x86_64-dell_s6100"
+
+    if [[ $ONIE_PLATFORM =~ $_DELLEMC_S6100_PLATFORM ]]; then
+        return 0
+    fi
+
+    return 1
+}
+
+function isMLNX() {
+    local -r _MELLANOX_PLATFORM="mlnx"
+
+    if [[ "${ONIE_SWITCH_ASIC}" = "${_MELLANOX_PLATFORM}" ]]; then
+        return 0
+    fi
+
+    return 1
+}
+
+
+function isArista() {
+    local -r _ARISTA_PLATFORM="x86_64-arista"
+
+    if [[ "${ABOOT_PLATFORM}" =~ "${_ARISTA_PLATFORM}" ]]; then
+        return 0
+    fi
+
+    return 1
+}
+
+
+
+function disable_watchdog()
+{
+    if isDellS6100; then
+        if [[ -x /usr/local/bin/platform_watchdog_disable.sh ]]; then
+            debug "Disabling Watchdog..."
+            /usr/local/bin/platform_watchdog_disable.sh
+        fi
+    fi
+    if isMLNX; then
+        debug "Disabling Watchdog..."
+        # call watchdog api for mlnx
+    fi
+    if isArista; then
+	if [[ -x /usr/bin/arista ]]; then
+            debug "Disabling Watchdog..."
+            /usr/bin/arista watchdog --stop
+        fi
+    fi
+}
+
+function enable_watchdog()
+{
+    if isDellS6100; then
+        if [[ -x /usr/local/bin/platform_watchdog_enable.sh ]]; then
+            debug "Enabling Watchdog..."
+	    /usr/local/bin/platform_watchdog_enable.sh
+        fi
+    fi
+    if isMLNX; then
+        debug "Enabling Watchdog..."
+        # call watchdog api for mlnx
+    fi
+    if isArista; then
+	if [[ -x /usr/bin/arista ]]; then
+            debug "Enabling Watchdog..."
+            /usr/bin/arista watchdog --arm $ARISTA_WD_TIMER
+        fi
+    fi
+}
+
+SCRIPT=$0
+
+function show_help_and_exit()
+{
+    echo "Usage ${SCRIPT} [options]"
+    echo "    Request Enabling/Disabling Watchdog. Invoke platform-specific tool when available."
+    echo " "
+    echo "    Available options:"
+    echo "        -h, -? : getting this help"
+    echo "        -e     : enabling platform hw watchdog"
+    echo "        -d     : disabling platform hw watchdog"
+
+    exit 0
+}
+
+function parse_options()
+{
+    while getopts "h?ved" opt; do
+        case ${opt} in
+            h|\? )
+                show_help_and_exit
+                ;;
+            v )
+                VERBOSE=yes
+                ;;
+            e )
+                ACTION=$ENABLE
+                ;;
+            d )
+                ACTION=$DISABLE
+                ;;
+        esac
+    done
+}
+
+parseMachineConf
+
+parse_options $@
+
+# Exit if not superuser
+if [[ "$EUID" -ne 0 ]]; then
+    echo "This command must be run as root" >&2
+    exit 1
+fi
+
+if [[ x"${ACTION}" == x"${ENABLE}" ]]; then
+    enable_watchdog
+fi
+
+if [[ x"${ACTION}" == x"${DISABLE}" ]]; then
```

Comment:
Since these `if` statements are mutually exclusive, I suggest changing this to an `elif`. You could then add an `else` statement which prints and/or logs an error/usage statement.